there are still some UI things to be done but i think its time for pull request @nejra

mulicvedad opened the pull request:
-resolved problems with gradient and shadows
-updated UI in order to fulfill demands from sketch
-rating with stars is now real (it was hardcoded before along with the
release date)
-did changes that my mentor suggested
still to do:
-replace standard UISegmentControl with the one that is specified in
project description (sketch)
-download more details about each movie (for example duration)

mulicvedad committed:
saved data locally with NSUserDefault and made a lot of smaller corrections of the existing code

-not ready yet for pull request, but should be by the end of monday

mulicvedad committed:
finished most functionalities

-resolved problems with gradient and shadows
-updated UI in order to fulfill demands from sketch
-rating with stars is now real (it was hardcoded before along with the
release date)
-did changes that my mentor suggested
still to do:
-replace standard UISegmentControl with the one that is specified in
project description (sketch)
-download more details about each movie (for example duration)

nejra commented:
You are getting this error on startup:

> This application is modifying the autolayout engine from a background thread after the engine was accessed from the main thread. This can lead to engine corruption and weird crashes.

Please, investigate this and when you fix it you won't have issues with slow news feed load :)

nejra commented on MovieApp/MovieApp/MoviesViewController.m:
```diff
@@ -19,8 +26,17 @@ @implementation MoviesViewController
+    downloader = [[MovieDBDownloader alloc] init];
+    [downloader configure];
+    [self.sortSegmentedControl setSelectedSegmentIndex:2];
+    [self.sortSegmentedControl sendActionsForControlEvents:UIControlEventValueChanged];
```
Add left and right padding to segmented control also (same as for collection view). You can leave it like this, since it's native design.

nejra commented:
Please, remove right navigation bar item, you won't be needing it. Also, put menu icon on the left (without any action for now).

nejra commented on MovieApp/MovieApp/MovieDBDownloader.h:
```diff
@@ -1,24 +1,19 @@
+    
 }Criterion;
 
 @interface MovieDBDownloader : NSObject
```
Probably should be called something like DataProviderService, since not only movies fill be fetched, nor just downloaded (there will be PUT, POST and DELETE requests too).

mulicvedad replied:
I called it that way because the api we are using is for "themoviedb", but the word downloader doesn't fit at all. I will rename it.

nejra commented on MovieApp/MovieApp/Movie.m:
```diff
@@ -4,4 +4,42 @@
+        self.releaseDate=[aDecoder decodeObjectForKey:@"release_date"];
+        self.posterPath=[aDecoder decodeObjectForKey:@"poster_path"];
+    }
+    return self;
```
Didn't we agree to remove these methods for coding/decoding object?

mulicvedad replied:
yes we did, but i forgot. my bad

nejra commented on MovieApp/MovieApp/MovieAppConfiguration.h:
```diff
@@ -1,16 +1,10 @@
 +(NSURL *)getFeedsSourceUrlPath;
 +(NSString *)getApiKey;
++(NSString *)getMoviesCollectionViewCellNibName;
++(NSString *)getMoviesCollectionViewCellIdentifier;
```
Nib name and cell identifier could've been the same, but ok.

nejra commented on MovieApp/MovieApp/MoviesViewController.m:
```diff
@@ -19,8 +26,17 @@ @implementation MoviesViewController
+    [self configureView];
+
+    downloader = [[MovieDBDownloader alloc] init];
+    [downloader configure];
```
Configuration part should be moved to AppDelegate and initialized on app startup only once for all mappings.

nejra replied:
Configuration part should be moved to AppDelegate and initialized on app startup only once for all mappings.  Object manager is shared object manager and you should make this class utility class (with all class methods). Let me know if there is some reason you can't do this.

nejra commented on MovieApp/MovieApp/MoviesViewController.m:
```diff
@@ -37,51 +53,87 @@ - (void)viewDidLoad {
+
+    cell.starsLabel.text=@"";
+    NSMutableString *stars=[[NSMutableString alloc]init];
+    NSUInteger numberOfStars=[self numberOfStarsFromRating:currentMovie.voteAverage];
```
numberOfStars can be calculated inside cell that is responsible for presenting this info (MoviesCollectionViewCell)

nejra commented on MovieApp/MovieApp/MoviesViewController.m:
```diff
@@ -37,51 +53,87 @@ - (void)viewDidLoad {
-{
-    CGFloat cellWidth=collectionView.bounds.size.width/2-[MoviesCollectionViewCell cellInsets].left*2 - [MoviesCollectionViewCell cellInsets].right*2;
+    }
+    cell.starsLabel.text=stars;
```
all related to setting `starsLabel` should be moved to `MoviesCollectionViewCell`

mulicvedad committed:
-fixed bug in loading News Fedd (dispatch_async…)
-updated UI based on latest sketch file that i got from our design
intern Jasna
-made first steps in making MoviesViewController more generic so i
could reuse code for TVShows collection view
-corrected other things that my mentor @nejra mentioned

nejra commented on MovieApp/MovieApp/DataProviderService.h:
```diff
@@ -0,0 +1,19 @@
+@interface DataProviderService : NSObject
+extern NSArray *niz;
+-(void)configure;
+-(void)getdMoviesByCriterion:(Criterion)criterion returnToHandler:(id<ItemsArrayReceiver>)delegate;
```
`configure`  and `getdMoviesByCriterion:returnToHandler:` should be class methods (or DataProviderService should be used as singleton, in order not to repeat configuration data like base url path)

nejra commented on MovieApp/MovieApp/MoviesViewController.m:
```diff
@@ -19,69 +26,76 @@ @implementation MoviesViewController
+    
+    //hardcoded for now
+    cell.genreLabel.text = @"Mistery, Thriller";
+    cell.ratingLabel.text=[NSString stringWithFormat:@"%.1f", currentMovie.voteAverage];
```
Please, add setup method for cell configuration to MoviesCollectionViewCell, like you did with news feed cell. This is practice that should be consistent!

nejra commented on MovieApp/MovieApp/MoviesCollectionViewCell.h:
```diff
@@ -1,21 +1,16 @@
 +(UIEdgeInsets)cellInsets;
 +(CGFloat)cellHeight;
 +(NSString *)cellIdentifier;
++(NSUInteger)numberOfStarsFromRating:(float)popularity;
```
As far as I can see, this method is unused.

mulicvedad committed:
almost done with lists of movies and tvshows

-made changes to MoviesViewController in order to make it more generic
and then used it as TvEventsCollectionViewController for both movies
and tv shows
-made changes in DataServiceProvider so i could use one method for
fetching list of movies and tvshows because these two are very similar
-updated fonts, colors and constraints
-still to do:
a)display genres
b)discuss with my mentor about segmented control because this solution
is ok for movies but  not apropriate for tvshows
